examples/api/lib/widgets/inherited_model/inherited_model.0.dart

```diff
@@ -0,0 +1,169 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for InheritedModel
+
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+
+void main() => runApp(const InheritedModelApp());
+
+class InheritedModelApp extends StatelessWidget {
+  const InheritedModelApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      home: InheritedModelExample(),
+    );
+  }
+}
+
+class FooBarModel extends InheritedModel<String> {
```

Comment:
Could we come up with a slightly more relevant/engaging use case? Foo and bar are very abstract and I got those two mixed up while reading through this example.